File: backend/src/models/responses/Brand.py
from typing import List, Optional
from pydantic import BaseModel
from uuid import UUID
from src.models.responses.Goods import GoodsModel

class BrandModel(BaseModel):
	uuid: UUID
	brand_name: str = ""
	telegram_id: str = ""
	brand_description: Optional[str] = ""
	social_links: Optional[List[str]] = []
	brand_logo_uuid: UUID
	brand_banners_uuid: UUID

class BrandResponse(BaseModel):
	brand: BrandModel
	goods: List[GoodsModel]

File: frontend/src/entities/Brand.ts
import { Goods } from "@entities";

export interface Brand {
	uuid: string; // uuid
	telegram_id: string;
	brand_name: string;
	brand_description?: string;
	social_links?: string[];
	brand_logo_uuid: string; // uuid
	brand_banners_uuid: string; // uuid
}

export interface BrandData {
	brand?: Brand
	goods: Goods[]
}